system_tests/bigquery.py

```diff
@@ -0,0 +1,220 @@
+        dataset.create()
+        self.to_delete.append(dataset)
+        TABLE_NAME = 'test_table'
+        full_name = bigquery.SchemaField('full_name', 'STRING', mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        table = dataset.table(TABLE_NAME, schema=[full_name, age])
+        self.assertFalse(table.exists())
+        table.create()
+        self.to_delete.insert(0, table)
+        self.assertTrue(table.exists())
+        self.assertEqual(table.friendly_name, None)
+        self.assertEqual(table.description, None)
+        table.patch(friendly_name='Friendly', description='Description')
+        self.assertEqual(table.friendly_name, 'Friendly')
+        self.assertEqual(table.description, 'Description')
+
+    def test_update_table(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        TABLE_NAME = 'test_table'
+        full_name = bigquery.SchemaField('full_name', 'STRING', mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        table = dataset.table(TABLE_NAME, schema=[full_name, age])
+        self.assertFalse(table.exists())
+        table.create()
+        self.to_delete.insert(0, table)
+        self.assertTrue(table.exists())
+        voter = bigquery.SchemaField('voter', 'BOOLEAN', mode='NULLABLE')
+        schema = table.schema 
+        schema.append(voter)
+        table.schema = schema
+        table.update()
+        self.assertEqual(len(table.schema), len(schema))
+        for found, expected in zip(table.schema, schema):
+            self.assertEqual(found.name, expected.name)
+            self.assertEqual(found.field_type, expected.field_type)
+            self.assertEqual(found.mode, expected.mode)
+
+    def test_load_dump_table(self):
+        ROWS = [
+            ('Phred Phlyntstone', 32),
+            ('Bharney Rhubble', 33),
+            ('Wylma Phlyntstone', 29),
+            ('Bhettye Rhubble', 27),
+        ]
+        ROW_IDS = range(len(ROWS))
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        TABLE_NAME = 'test_table'
+        full_name = bigquery.SchemaField('full_name', 'STRING', mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        table = dataset.table(TABLE_NAME, schema=[full_name, age])
+        self.assertFalse(table.exists())
+        table.create()
+        self.to_delete.insert(0, table)
+        self.assertTrue(table.exists())
+        
+        errors = table.insert_data(ROWS, ROW_IDS)
+        self.assertEqual(len(errors), 0)
+
+        rows = ()
+        counter = 9
+        # Allow for 90 seconds of "warm up" before rows visible.  See:
```

Comment:
Yup.  And it does occur in the wild:  the test fails repeatably without the sleep.